### ISSUE TITLE
Indentation check fails for Line Wrapped Anonymous Inner Class with LCurly on newline

### ISSUE THREAD

**Issue by JLLeitschuh** [checkstyle/checkstyle#3612]
````
$ cat TestClass.java
public class TestClass
{
    public static class SimpleClass {
        public void method() {}
    }

    void method()
    {
        final SimpleClass test =
                new SimpleClass() { //indent:16
                    public void method() {}
                }; //indent:16
        final SimpleClass test2 =
                new SimpleClass() //indent:16
                { //indent:16 violation
                    public void method() {}
                }; //indent:16
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="Indentation">
  <property name="tabWidth" value="4" />
  <property name="basicOffset" value="4" />
  <property name="braceAdjustment" value="0" />
  <property name="caseIndent" value="4" />
  <property name="lineWrappingIndentation" value="8" />
  <property name="throwsIndent" value="4" />
  <property name="arrayInitIndent" value="4" />
</module>
    </module>
</module>

$ java -jar checkstyle-7.4-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:15: 'object def lcurly' have incorrect indentation level 16, expected level should be one of the following: 8, 12. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
````

There are two test cases here, the first, `test` passes because the `LCURLY` is on the same line.
`test2` fails because the `LCURLY` is on its own line alone.

the current validation code wants this:
```java
        final SimpleClass test2 =
                new SimpleClass() //indent:16
            { //indent:12
                public void method() {}
            }; //indent:12
```

https://github.com/JLLeitschuh/checkstyle/commit/661ad249fdb723b12e4c95cbfdf8f758c7465faa#diff-41124079fe0fabb5d72bbb58a1f17c8c
Above has test case and fix.

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/39791441-indentation-check-fails-for-line-wrapped-anonymous-inner-class-with-lcurly-on-newline?utm_campaign=plugin&utm_content=tracker%2F500528&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F500528&utm_medium=issues&utm_source=github).

**Comment by JLLeitschuh** [comment]
You can see the attached test, I'm willing to fix this if someone would be so kind as to direct me where to look to fix this problem.

**Comment by romani** [comment]
@rnveach , please assist.

**Comment by JLLeitschuh** [comment]
I think I solved it.

**Comment by rnveach** [comment]
@JLLeitschuh Is the configuration in the PR the one you used to get the violation?

````
        checkConfig.addAttribute("tabWidth", "4");
        checkConfig.addAttribute("basicOffset", "4");
        checkConfig.addAttribute("braceAdjustment", "0");
        checkConfig.addAttribute("caseIndent", "4");
        checkConfig.addAttribute("lineWrappingIndentation", "8");
        checkConfig.addAttribute("throwsIndent", "4");
        checkConfig.addAttribute("arrayInitIndent", "4");
````

**Comment by JLLeitschuh** [comment]
Yes

**Comment by rnveach** [comment]
@romani The issue looks valid to me. Braces should be on the same identation for their respective lines.

The issue shows 2 similar cases, with the only difference is where the opening brace is located.
Similified version:
````
                new SimpleClass() { //indent:16
                }; //indent:16
                new SimpleClass() //indent:16
                { //indent:16 violation
                }; //indent:16
````

The line for the braces are at the same indentation (both 16), and we accept the location of the `new Class`, so I don't think there should be any violation in the example.

Current violation wants the code to look like:
````
                new SimpleClass() { //indent:16
                }; //indent:16
                new SimpleClass() //indent:16
            { //indent:12
                }; //indent:16
````
Which definitely looks wrong.

**Comment by JLLeitschuh** [comment]
Actually, the current validation code wants this:
```java
                new SimpleClass() //indent:16
            { //indent:12
            }; //indent:12
```

**Comment by romani** [comment]
@JLLeitschuh,  please report issue as was described in issue template - https://github.com/checkstyle/checkstyle/blob/master/.github/ISSUE_TEMPLATE.md
it do helps to for investigation now and for future investigations.
Please minimize Input file to let see problem easily as @rnveach did.

After that issue could be marked as "approved".

**Comment by romani** [comment]
issue need to be reported properly. 
issue description need to be updated to clearlt state the problem.
After isuue is approved, code should be reviweved, regression testing report should be provided (ut is not enouth).

**Comment by rnveach** [comment]
@romani Here is simplified case:
````
$ cat TestClass.java
public class TestClass
{
    public static class SimpleClass {
        public void method() {}
    }

    void method()
    {
        final SimpleClass test =
                new SimpleClass() { //indent:16
                    public void method() {}
                }; //indent:16
        final SimpleClass test2 =
                new SimpleClass() //indent:16
                { //indent:16 violation
                    public void method() {}
                }; //indent:16
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="Indentation">
  <property name="tabWidth" value="4" />
  <property name="basicOffset" value="4" />
  <property name="braceAdjustment" value="0" />
  <property name="caseIndent" value="4" />
  <property name="lineWrappingIndentation" value="8" />
  <property name="throwsIndent" value="4" />
  <property name="arrayInitIndent" value="4" />
</module>
    </module>
</module>

$ java -jar checkstyle-7.4-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:15: 'object def lcurly' have incorrect indentation level 16, expected level should be one of the following: 8, 12. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
````

**Comment by romani** [comment]
fix is merged.
